firewall/handlers/blacklist_handler.go

```diff
@@ -0,0 +1,58 @@
+package handlers
+
+import (
+	"errors"
+
+)
+
+type BlacklistHandler struct {
+	blackQueries[] string
+}
+
+var ErrQueryInBlacklist = errors.New("query in blacklist")
+
+func NewBlacklistHandler(blackQueries []string) (*BlacklistHandler, error) {
+
+	return &BlacklistHandler{blackQueries:blackQueries}, nil
+}
+
+func(handler * BlacklistHandler) CheckQuery(query string) error {
+	yes, _ := handler.contains(query)
+	if yes {
+		return ErrQueryInBlacklist
+	}
+	return nil
+}
+
+func(handler * BlacklistHandler) AddQueriesToBlacklist(queries []string) {
+
+	for index := 0; index < len(queries); index++ {
+		handler.blackQueries = append(handler.blackQueries, queries[index])
+	}
+	removeDuplicates(&handler.blackQueries)
+
+
+}
+
+func(handler * BlacklistHandler) RemoveQueriesFromBlacklist(queries []string){
+
+	for i := 0; i < len(queries); i++ {
+		yes, index := handler.contains(queries[i])
+		if yes {
+			handler.blackQueries = append(handler.blackQueries[:index], handler.blackQueries[index+1:]...)
+		}
+	}
+}
+
+func(handler * BlacklistHandler) contains(query string) (bool, int) {
```

Comment:
this function used twice in two handlers
what about to make it as simple function like `contains(query string, queries[]string)(bool, int)` and use it in handlers? then we need test this function just once

Reply:
Agree. Moved to 'handlers' subpackage